File: python/SelectionSort.py
def selection_sort_ascended(a):
    for i in range(len(a)):
        min_key = i
        for j in range(i + 1, len(a)):
            if a[j] < a[min_key]:
                min_key = j
        a[i], a[min_key] = a[min_key], a[i]
        print("after %d change:"%i, a)
    return a

def selection_sort_dascended(a):
    for i in range(len(a)):
        max_key = i
        for j in range(i + 1, len(a)):
            if a[j] > a[max_key]:
                max_key = j
        a[i], a[max_key] = a[max_key], a[i]
        print("after %d change:"%i, a)
    return a

def main():
    Array = list(map(int,input("please input the array(split by space):").split()))
    print("input is: ", Array)
    SortedArrayAscend = selection_sort_ascended(Array)
    print("ascend result is: ", SortedArrayAscend)

    SortedArrayDascend = selection_sort_dascended(Array)
    print("dascend result is: ", SortedArrayDascend)


if __name__ == "__main__":
    main()

File: C++/SelectionSort.cpp
#include <iostream>
#include <stdlib.h>
#include <vector>

using namespace std;

void print_array(auto array)
{
    for(int i : array)
    {
        cout << i << " ";
    }
    cout << endl;
}

void selection_sort(auto &array)
{
    auto len = array.size();
    for (int i = 0; i < len; i++)
    {
        int key = i;
        for(int j = i+1; j < len; j++)
        {
            if(array[j] < array[key])
            {
                key = j;
            }
        }
        if(key != i)
        {
            int temp = array[i];
            array[i] = array[key];
            array[key] = temp;
        }

        cout << "after " << i + 1 << " change: ";
        print_array(array);
    }
}

int main()
{  
    vector<int> array;
    char ch;
    cout << "please input the array(split by space):";
    while((ch = cin.get()) != '\n')
    {
        if(ch != ' ')
            array.push_back(int(ch - '0'));
    }
    cout << "origin is: " ;
    print_array(array);

    selection_sort(array);
    cout << "result is: " ;
    print_array(array);

    system("pause");
    return 0;
}